### ISSUE TITLE
Proxy from Concourse is not set in Docker daemon  

### ISSUE THREAD

**Issue by michaellihs** [concourse/docker-image-resource#110]
We rolled out Concourse with the BOSH Release and set our Proxy via

```
instance_groups:
- name: worker
  jobs:
  - name: groundcrew
    release: concourse
    http_proxy_url: http_proxy=PROXY_URL
    https_proxy_url: https_proxy=PROXY_URL
    no_proxy: [...]
```

Within "normal" Concourse tasks the proxy works as expected, e.g. we can download files with `wget`.

We now try to build a Docker image from a Dockerfile with the following configuration:

```
---
resources:
- name: concourse-dockerfile
  type: git
  source:
    uri: ssh://git@XXX/bosh/concourse-dockerfile.git
    private_key: (( grab meta.ssh_key ))

- name: concourse-docker-image
  type: docker-image
  source:
    repository: DOCKER-REGISTRY/osc1/concourse
    username: {{artifactory_user}}
    password: {{artifactory_password}}
    insecure_registries:
    - DOCKER_REGISTRY

jobs:
- name: build-rootfs
  plan:
  - get: concourse-dockerfile 
  - put: concourse-docker-image
    params: {build: concourse-dockerfile}
```

When we now run this pipeline, we get the following error message:

```
Login Succeeded
Sending build context to Docker daemon 70.14 kB
Step 1/16 : FROM ubuntu
Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
```

which I think means, that the Docker daemon cannot access the Docker Registry, hence the proxy is not set for Docker.

What we do with other Docker daemons that we have set up, is to add the following line in the `/etc/systemd/system/docker.service.d/http-proxy.conf` file:

```
[Service]
Environment="HTTP_PROXY=http://PROXY_URL:8080"
```

Question is, how this can be achieved with Concourse / the Docker image resource.

**Comment by chendrix** [comment]
@michaellihs Is this something that would be fixed by concourse/concourse#1145?

**Comment by chendrix** [comment]
Actually, I think this should be done via garden's setting of [http/s_proxy](http://bosh.io/jobs/garden?source=github.com/cloudfoundry/garden-runc-release&version=1.8.0#p=garden.http_proxy)

According to the docs I'm reading about the docker daemon, it should pull those variables from the environment when starting up the daemon.
